core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -479,25 +479,33 @@ class LogManager(logDirs: Seq[File],
 
     try {
       for ((dir, dirJobs) <- jobs) {
-        dirJobs.foreach(_.get)
+        val hasErrors = dirJobs.exists {
```

Comment:
This looks wrong. `exists` short-circuits. I think you want `map` followed by `exists`.

Reply:
Thats a really good point. Done.